internal/build/lifecycle_execution.go

```diff
@@ -707,6 +722,50 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, buildCache Cache, ph
 	return extend.Run(ctx)
 }
 
+func (l *LifecycleExecution) ExtendRunByDaemon(ctx context.Context, currentRunImage *string) error {
+	defaultFilterFunc := func(file string) bool { return true }
+	var extensions Extensions
+	l.logger.Debugf("extending run image %s", *currentRunImage)
+	fmt.Println("tmpDir: ", l.tmpDir)
+	extensions.SetExtensions(l.tmpDir, l.logger)
+	dockerfiles, err := extensions.DockerFiles(DockerfileKindRun, l.tmpDir, l.logger)
+	if err != nil {
+		return fmt.Errorf("getting %s.Dockerfiles: %w", DockerfileKindRun, err)
+	}
+	fmt.Println("Dockerfiles: ", dockerfiles)
+	fmt.Println("extend: ", dockerfiles[1].Extend)
+	for _, dockerfile := range dockerfiles {
+		if dockerfile.Extend {
+			fmt.Println("dockerfile: ", dockerfile)
+			fmt.Println("dockerfile.Path dir: ", filepath.Dir(dockerfile.Path))
+			buildContext := archive.ReadDirAsTar(filepath.Dir(dockerfile.Path), "/", 0, 0, -1, true, false, defaultFilterFunc)
+			fmt.Println("buildContext: ", buildContext)
+			buildArguments := map[string]*string{}
+			if dockerfile.WithBase == "" {
+				buildArguments["base_image"] = currentRunImage
+			}
+			buildOptions := types.ImageBuildOptions{
+				Context:    buildContext,
+				Dockerfile: "Dockerfile",
+				Tags:       []string{"run-image"},
```

Comment:
We probably need to come up with some unique identifier here

Reply:
fixed in the latest commit